R/evaluate-promise.R

```diff
@@ -39,9 +39,12 @@ evaluate_promise <- function(code, print = FALSE) {
   temp <- file()
   on.exit(close(temp))
 
-  result <- with_sink(temp,
-    withCallingHandlers(
-      withVisible(code), warning = wHandler, message = mHandler
+  with_reporter(
```

Comment:
The second warning wouldn't be run with tryCatch():

```
> tryCatch({warning(1); warning(2)}, warning = identity)$message
[1] "1"
```

Can't use withCallingHandlers() because this doesn't eat warnings (or I'd have to reimplement most of the logic in test_code() ).

Am I missing anything obvious?